### DISCUSSION TITLE
sparse cloud from known poses

### DISCUSSION THREAD

**Discussion by avispiel** [openMVG/openMVG#2076] [General]
Hi.
I've used my images with known telemetry and created my own implementation of SfM_Data_GT_Loader_Interface
I have what seems to be a valid sfm_data.json
Then I do matching, reconstruct sequential and reconstruction global.
Shouldn't  I expect the coordinates of the structure vertices to comply with the sfm_data.json data?
the sfm_adta.json contains real world geocentric coordinates and rotations,
shouldn't I expect a geocentric coordinate cloud?
Thank you!
[sfm_data.json.txt](https://github.com/openMVG/openMVG/files/8813345/sfm_data.json.txt)

**Comment by pmoulon** [comment]
Hello.
Your JSON file seems ok.
Did you try to plot the camera using the binary exportCameraFrustum?

- Rotations must be from world to cam.

In order to use your pose I would advise you to use SFM with Incremental V2 and use pose mode. 
- check that you have pairwise correspondences (you can use exportMatches binary)

**Reply by avispiel** [reply]
I used to run first INCREMENTAL and then global.
I tried running only INCREMENTALV2 instead because I saw it has an option of  --sfm_initializer EXISTING_POSE
but the cloud turns out much worse.

Now Lets say I keep the original flow and get the sfm_data in some local coordinate system.
I compute a similarity transform I want to apply and move + ratate + scale the whole scene.
poses and cloud point are easy - I just apply the transform.
what do I do with the rotation?
should I be worried about the intrinsics?
Thank you!

**Comment by avispiel** [comment]
Hi. 
An update if someone is interested.

The results INCREMENTALV2 produces in my use cases are much worth so I looked for another  solution.

I found a workaround.
I used the file main_geodesy_registration_to_gps_position.
It constructs a target vector of positions using exif data from images (I dont have exif data in my images)
and constructs a source vector of positions from the sfm_data.
then it computes a 3d similarity between these 2 vector
then it applies this similarity to the sfm_data (structure, poses, cameras)

I only modified the extraction of the destination vector of positions to extract from my data instead of exif data.
I found and applied the transform after the whole process was complete.
The result was as expected.

**Reply by pmoulon** [reply]
Happy that you were able to find a solution to your problem
